fvm/evm/events/events.go

```diff
@@ -62,13 +64,21 @@ func (p *transactionEvent) ToCadence(chainID flow.ChainID) (cadence.Event, error
 
 	eventType := stdlib.CadenceTypesForChain(chainID).TransactionExecuted
 
+	errorMessage := p.Result.ErrorMsg()
+	if p.Result.ResultSummary().ErrorCode == types.ExecutionErrCodeExecutionReverted {
+		reason, errUnpack := abi.UnpackRevert(p.Result.ReturnedData)
+		if errUnpack == nil {
+			errorMessage = fmt.Sprintf("%v: %v", gethVM.ErrExecutionReverted.Error(), reason)
+		}
+	}
```

Comment:
Can you make this a method on `types.Result`. Maybe call it `ErrorMessageWithRevertReason` or something.

Reply:
Nice idea :ok_hand: Updated in https://github.com/onflow/flow-go/pull/6836/commits/e682cd2dc8ee35aeafadb5b179b280253f86f4c5 .